commit fbffee74986c899524de5707dab9cf321d02dab8
Author: Kalyan Kinthada <kalyan.kinthada@alliedtelesis.co.nz>
Date:   Fri Sep 29 10:36:04 2017 +1300

    ARM: dts: Fix I2C repeated start issue on Armada-38x
    
    All the Armada 38x(380, 385, 388) have a silicon issue in
    the I2C controller which violates the I2C repeated start timing
    (errata FE-8471889).
    
    i2c-mv64xxx driver handles this errata based on the compatible string
    "marvell,mv78230-a0-i2c".
    
    This patch activates the "marvell,mv78230-a0-i2c" compatible string
    for the I2C controller on armada-38x SoC based devices.
    
    Signed-off-by: Kalyan Kinthada <kalyan.kinthada@alliedtelesis.co.nz>
    Signed-off-by: Gregory CLEMENT <gregory.clement@free-electrons.com>

diff --git a/arch/arm/boot/dts/armada-38x.dtsi b/arch/arm/boot/dts/armada-38x.dtsi
index 7ff0811e..4960722 100644
--- a/arch/arm/boot/dts/armada-38x.dtsi
+++ b/arch/arm/boot/dts/armada-38x.dtsi
@@ -178,7 +178,7 @@
 			};
 
 			i2c0: i2c@11000 {
-				compatible = "marvell,mv64xxx-i2c";
+				compatible = "marvell,mv78230-a0-i2c", "marvell,mv64xxx-i2c";
 				reg = <0x11000 0x20>;
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -189,7 +189,7 @@
 			};
 
 			i2c1: i2c@11100 {
-				compatible = "marvell,mv64xxx-i2c";
+				compatible = "marvell,mv78230-a0-i2c", "marvell,mv64xxx-i2c";
 				reg = <0x11100 0x20>;
 				#address-cells = <1>;
 				#size-cells = <0>;
